### PR TITLE
change to generate kustomize base files from helm chart

### LINKED ISSUE #1546: change to generate kustomize base files from helm chart
**Problem**
The current chart is generated by `hack/generate-helm.sh`, which makes it difficult to modify the Helm template.

Therefore, I propose generating the base files used for Kustomize from the chart, in the opposite way of the current method.

**[Persona:](https://github.com/knative/eventing/blob/main/docs/personas.md)**
System Operator

**Exit Criteria**
Pass existing CI tests using helm chart and kustomize

**Time Estimate (optional):**
1 developer, 1~2 days

### After changes

```shell
./
├── config/
│   ├── charts/
│   │   ├── knative-operator/
│   │   │   ├── Chart.yaml
│   │   │   ├── templates/
│   │   │   │   ├── crd/
│   │   │   │   │   └── bases/
│   │   │   │   │       ├── operator.knative.dev_knativeeventings.yaml
│   │   │   │   │       └── operator.knative.dev_knativeservings.yaml
│   │   │   │   ├── manager/
│   │   │   │   │   ├── config-logging-configmap.yaml
│   │   │   │   │   ├── config-observability-configmap.yaml
│   │   │   │   │   └── operator.yaml
│   │   │   │   ├── rbac/
│   │   │   │   │   ├── clusterrole_aggregated.yaml
│   │   │   │   │   ├── clusterrole_aggregated_binding.yaml
│   │   │   │   │   ├── fake_clusterrole_aggregated_binding.yaml
│   │   │   │   │   ├── fake_service_account.yaml
│   │   │   │   │   ├── role.yaml
│   │   │   │   │   ├── role_binding.yaml
│   │   │   │   │   ├── service_account.yaml
│   │   │   │   │   ├── webhook_role.yaml
│   │   │   │   │   ├── webhook_role_binding.yaml
│   │   │   │   │   └── webhook_service_account.yaml
│   │   │   │   ├──  webhook/
│   │   │   │   │   ├── webhook-secret.yaml
│   │   │   │   │   ├── webhook-service.yaml
│   │   │   │   │   └── webhook.yaml
│   │   │   │   └──  ...
│   │   │   └── values.yaml
│   │   └──  ...
│   ├── crd/
│   │   └──  ...
│   ├── manager/
│   │   └──  ...
│   ├── rbac/
│   │   └──  ...
│   ├── webhook/
│   │   └──  ...
│   └──  ...
├── hack/
│   ├── generate-kustomize.sh
│   └──  ...
└──  ...
```

### PR DESCRIPTION
**Pull request by hhk7734**
Fixes #1546

Hi!
Changed Helm templates for more flexiblity and added scripts to generate kustomize default files from charts.

When I compared the operator-hub bundles before and after changes, I got the below.

```diff
diff -r bundle_old bundle
diff --color -r bundle_old/manifests/knative-operator.v1.6.0.clusterserviceversion.yaml bundle/manifests/knative-operator.v1.6.0.clusterserviceversion.yaml
28c28
<     createdAt: "2023-11-23T13:22:01Z"
---
>     createdAt: "2023-11-23T15:59:22Z"
```

### CODE REVIEW

**Comment by tynfojlyue** [comment]
I think this is a great change.

**Comment by houshengbo** [comment]
@hhk7734 Would you mind trying to fix the CI build issues for this PR?

**Comment by hhk7734** [comment]
I just rebased to origin/main without changing any code. :)

File: config/charts/knative-operator/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: knative-operator
-version: {{ version }}
+version: 1.10.1
```
Comment (houshengbo) [inline config/charts/knative-operator/Chart.yaml:3]: Why did you pin a specific version? I think we would like to release the charts based on different versions.

File: config/charts/knative-operator/values.yaml [hunk 2]
```diff
@@ -1,8 +1,10 @@
+namespaceOverride: ""
+
 knative_operator:
   knative_operator:
     image: gcr.io/knative-releases/knative.dev/operator/cmd/operator
-    tag: {{ tag }}
+    tag: v1.10.1
```
Comment (houshengbo) [inline config/charts/knative-operator/values.yaml:6]: Why do you pin a specific version?

**Comment by ryan-dyer-sp** [comment]
Can you please include the ability to add custom annotations to all of the objects?  Our use case would be to bundle the operator with default knativeserving/eventing objects along with the kafka eventing controller within an argocd application and we would need to control install order via argo's sync-wave annotations.  The kafka eventing controller does not offer a helm chart so we only have a manifest file which would run at a default order of 0.  So the ability to say the operator's sync-wave itself is <0 would be great.

**Review by ryan-dyer-sp — COMMENTED** [review]
There are still a few `namespace: default` references.  Shouldnt these all be `namespace: {{ include "knativeOperator.namespace" .}}`

**Pull request closed without merging**